src/promptflow/promptflow/integrations/langchain.py

```diff
@@ -96,7 +134,8 @@ def on_chain_end(self, outputs: Dict[str, Any], **kwargs: Any) -> None:
         :type outputs: Dict[str, Any]
         """
 
-        self._pop(outputs)
+        if self._try_pop_event_type(LangChainEventType.CHAIN):
+            self._pop(outputs)
```

Comment:
Shall we embed the `if` logic into the `self._pop()` method so that this line can kept unchanged and easy to understand?

Reply:
Refined.